File: tcpcom/PC/Python/Examples/TestClient.py
# TestServer.py

from tcpcom import TCPClient
from easygui import msgbox  # Standard Python with easygui

def onStateChanged(state, msg):
    print state, "-", msg

host = "localhost"
port = 5000

client = TCPClient(host, port, stateChanged = onStateChanged, isVerbose = True)
success = client.connect()
if success:
    msgbox("Client connected. OK to disconnect","Test Client")  # Standard Python
    client.sendMessage("Test Client says Hello To You!")
    client.disconnect()
else:
    msgbox("Client connection failed. OK to quit","Test Client")   # Standard Python

File: tcpcom/PC/Java/Examples/TestClient.java
// TestClient.java

import ch.aplu.tcpcom.*;
import ch.aplu.util.*;

public class TestClient implements TCPClientListener
{
  private TCPClient client;
  private Console c;

  public TestClient()
  {
    c = new Console();
    c.setTitle("Test Client");
    System.out.println("Trying to connect..");
    String host = "localhost";
    int port = 5000;
    client = new TCPClient(host, port, true);
    client.addTCPClientListener(this);
    boolean success = client.connect();
    if (success)
    {
      System.out.println("Connection successful. Close to disconnect");
//      client.sendMessage("Test Client says hello to you!");
      while (!c.isDisposed())
        Thread.yield();
      client.disconnect();
    }
    else
      System.out.println("Connection failed");
  }

  public void onStateChanged(String state, String msg)
  {
    System.out.println(state + " - " + msg);
  }

  public static void main(String[] args)
  {
    new TestClient();
  }
}
